### ISSUE TITLE
Remote Config not being updated in realtime even if Realtime API is enabled

### ISSUE THREAD

**Issue by alvindizon** [firebase/firebase-android-sdk#4864]
* Android Studio version: Android Studio Electric Eel | 2022.1.1 Patch 1
  * Firebase Component: Remote Config Realtime API
  * Component version: 31.4.0 (Bill of Materials)

### [REQUIRED] Step 3: Describe the problem
Updates to Remote Config are not coming thru despite being enabled in Google Cloud.

#### Steps to reproduce:

1. I have an existing set of Remote Configs that I've updated via Firebase Console, but even after enabling the realtime API in Google Cloud and setting it up a listener for config changes on my Android app, my app did not receive any updates. Inspecting the graphs at Google Cloud, it seems that for the past hour, only errors (with status code 503) have been recorded, please see graphs:

![Traffic by response code](https://user-images.githubusercontent.com/30955662/230401047-f351e651-7cb7-4969-8597-45e1ce7af460.png)

![Errors by API method](https://user-images.githubusercontent.com/30955662/230401203-28be2e10-3d98-42f7-ad06-8d1232466871.png)

<img width="1236" alt="image" src="https://user-images.githubusercontent.com/30955662/230402222-2efeff40-7050-4ee9-b375-7b9448373863.png">

#### Relevant Code:

In my Application class, I initialize a class that wraps around FirebaseRemoteConfig:
```
        remoteConfigManager = new FirebaseRemoteConfigManager(FirebaseRemoteConfig.getInstance());
        remoteConfigManager.initializeConfig();
        remoteConfigManager.fetchLiveUpdates();
```

Within the FirebaseRemoteConfigManager class, `fetchLiveUpdates` calls `addOnConfigUpdateListener` and listens for updates:

```
class FirebaseRemoteConfigManager(private val firebaseRemoteConfig: FirebaseRemoteConfig): RemoteConfigManager {

    override fun initializeConfig() {
        Log.w(TAG, "Initialize remote config")
        with(firebaseRemoteConfig) {
            setConfigSettingsAsync(
                remoteConfigSettings {
                    minimumFetchIntervalInSeconds = 0L
                }
            )
            setDefaultsAsync(R.xml.remote_config_defaults)
        }
    }
    
       override fun fetchLiveUpdates() {
        Log.w(TAG, "Start live updates")
        firebaseRemoteConfig.addOnConfigUpdateListener(object : ConfigUpdateListener {
            override fun onUpdate(configUpdate: ConfigUpdate) {
                Log.w(TAG, "Updated keys: " + configUpdate.updatedKeys)
                firebaseRemoteConfig.activate()
            }

            override fun onError(error: FirebaseRemoteConfigException) {
                Log.w(TAG, "Config update error with code: ${error.code}")
            }
        })
    }
}
```

`onUpdate` and even `onError`were not triggered even if I published my Remote Configs.

**Comment by alvindizon** [comment]
Hi @argzdev , do you have a working project setup or sample code that can be used as a reference? I've followed the instructions [here](https://firebase.google.com/docs/remote-config/get-started?platform=android#add-real-time-listener), but as I've said and shown here, all fetches have resulted in 503 errors.

**Comment by argzdev** [comment]
There's really not much that I did, I created a clean project and then added the code snippet you've provided. Here's my [project setup](https://github.com/argzdev/issue4864/tree/main). You could also try using our [Firebase quickstart](https://github.com/firebase/quickstart-android/tree/master/config), and see if you can make it work first then add realtime API. But from what I can tell, there's probably something wrong with the connectivity with the server. Are you using AppCheck, perchance?

**Comment by alvindizon** [comment]
Thanks @argzdev for providing a reference project. We're not using AppCheck--I'll study your project and with the support ticket I opened to the Firebase support team, I'll wait for further updates. Thanks again

**Comment by alvindizon** [comment]
Hello @argzdev , I've used the Firebase quickstart and true enough, updates are being propagate at almost realtime (around 8 mins on my end). However, we are still getting no updates when we try to adapt it to our code (i.e., initialize and fetch remote config on Activity class). What's even more frustrating is that we're already seeing this kind of error on our logs:

```
Config update error with code: CONFIG_UPDATE_STREAM_ERROR
                 com.google.firebase.remoteconfig.FirebaseRemoteConfigClientException: Unable to connect to the server. Check your connection and try again.
                 	at com.google.firebase.remoteconfig.internal.ConfigRealtimeHttpClient.makeRealtimeHttpConnection(ConfigRealtimeHttpClient.java:361)
                 	at com.google.firebase.remoteconfig.internal.ConfigRealtimeHttpClient.retryHttpConnectionWhenBackoffEnds(ConfigRealtimeHttpClient.java:341)
                 	at com.google.firebase.remoteconfig.internal.ConfigRealtimeHttpClient.beginRealtimeHttpStream(ConfigRealtimeHttpClient.java:454)
                 	at com.google.firebase.remoteconfig.internal.ConfigRealtimeHttpClient$1.run(ConfigRealtimeHttpClient.java:355)
                 	at com.google.firebase.concurrent.DelegatingScheduledExecutorService.lambda$schedule$0(DelegatingScheduledExecutorService.java:116)
                 	at com.google.firebase.concurrent.DelegatingScheduledExecutorService$$ExternalSyntheticLambda11.run(Unknown Source:4)
                 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
                 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
                 	at com.google.firebase.concurrent.CustomThreadFactory.lambda$newThread$0$com-google-firebase-concurrent-CustomThreadFactory(CustomThreadFactory.java:47)
                 	at com.google.firebase.concurrent.CustomThreadFactory$$ExternalSyntheticLambda0.run(Unknown Source:4)
                 	at java.lang.Thread.run(Thread.java:1012)
```

**Comment by alvindizon** [comment]
Not to mention that even without using the realtime API, Firebase Remote Config seems to be very inconsistent and has issues with syncing values when used together with custom user properties, see the Stackoverflow question I posted for details: https://stackoverflow.com/questions/75945085/firebase-remote-config-values-not-consistent-after-user-properties-are-changed-o and my post over at the discussions tab: https://github.com/firebase/firebase-android-sdk/discussions/4872

Could you please guide us on the next steps? We're waiting for a response to our support ticket, and we'd really like to use Firebase Remote Config as much as possible.

**Comment by danasilver** [comment]
Hi! Sorry you're running into this @alvindizon and thanks for reporting the issue! We're taking a look and I'll update this issue with any information.

**Comment by danasilver** [comment]
Hi @alvindizon, we identified and fixed an issue on the backend that may have been causing this error. Could you try running your app again and let us know if you're still seeing this issue?

**Comment by alvindizon** [comment]
Hi @danasilver thanks for the update and sorry for the late reply, I will be checking it again today.

**Comment by alvindizon** [comment]
Hi @danasilver, I've observed that while the graphs at Cloud console show no more errors (200 status code), I've received no updates on my Android app.

![Traffic by response code](https://user-images.githubusercontent.com/30955662/231419026-23b31882-71a4-4a84-b726-638755a9bf25.png)


![Errors by credential](https://user-images.githubusercontent.com/30955662/231419106-78d5ad58-6376-4381-b315-4c9bd49914b3.png)


![Overall latency](https://user-images.githubusercontent.com/30955662/231419148-5f31f9b5-e80e-40dd-95f1-307facbfd16b.png)

<img width="1231" alt="Screenshot 2023-04-12 at 5 37 08 PM" src="https://user-images.githubusercontent.com/30955662/231419241-fdc3a8b7-c0b3-4c8b-b5af-44b235f443b6.png">

**Comment by alvindizon** [comment]
Hi @danasilver  and @argzdev, I think I've made a breakthrough. First, I noticed on Logcat this message: `Config was fetched, but no params changed.` Then I had the idea of modifying @argzdev 's project [here](https://github.com/argzdev/issue4864), copied the namespace, signing keystore and `google-services.json` that we use for our app, and I can confirm that I can receive updates from Firebase in realtime, however, in our actual production app, we are still not getting any updates. Then I noticed that in our prod app, we both use Java and Kotlin versions of the remote config library and the Firebase analytics library, whereas @argzdev 's project only uses Kotlin libraries. So I removed the Java version on our prod app, and then I tried changing the remote config values in Firebase, and voila, the updates came through!

I don't understand why this behavior appears when using both Java and Kotlin libraries, but I'll leave this info for you guys if you want to investigate further. Thanks for your assistance!
